tests/integration/test_s3.py

```diff
@@ -121,3 +218,146 @@ def test_s3_multipart(self):
 
         contents = bucket.Object('mp-test.txt').get()['Body'].read()
         self.assertEqual(contents, b'hello, world!')
+
+
+class TestS3Transfers(unittest.TestCase):
+    """Tests for the high level boto3.s3.transfer module."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.region = 'us-west-2'
+        cls.session = boto3.session.Session(region_name=cls.region)
+        cls.client = cls.session.client('s3', cls.region)
+        cls.bucket_name = 'boto3-integ-transfer-%s-%s' % (
+            int(time.time()), random.randint(1, 100))
+        cls.client.create_bucket(
+            Bucket=cls.bucket_name,
+            CreateBucketConfiguration={'LocationConstraint': cls.region})
+
+    def setUp(self):
+        self.files = FileCreator()
+
+    def tearDown(self):
+        self.files.remove_all()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.client.delete_bucket(Bucket=cls.bucket_name)
+
+    def delete_object(self, key):
+        self.client.delete_object(
+            Bucket=self.bucket_name,
+            Key=key)
+
+    def object_exists(self, key):
+        self.client.head_object(Bucket=self.bucket_name,
+                                Key=key)
+        return True
+
+    def create_s3_transfer(self):
+        return boto3.s3.transfer.S3Transfer(self.client)
+
+    def test_upload_below_threshold(self):
+        transfer = self.create_s3_transfer()
+        filename = self.files.create_file_with_size(
+            'foo.txt', filesize=1024 * 1024)
+        transfer.upload_file(filename, self.bucket_name,
+                             'foo.txt')
+        self.addCleanup(self.delete_object, 'foo.txt')
+
+        self.assertTrue(self.object_exists('foo.txt'))
```

Comment:
Should we also be asserting that the file size was below the configured threshold? It seems like we're making assumptions in the tests about defaults in the code that may change over time.